gateway/gateway.go

```diff
@@ -73,6 +73,15 @@ type webRequestT struct {
 	writeKey       string
 	ipAddr         string
 	userIDHeader   string
+	errors         []string
+}
+
+func (wr *webRequestT) respond() {
+	if len(wr.errors) > 0 {
+		wr.done <- wr.errors[0]
+	} else {
+		wr.done <- ""
+	}
```

Comment:
I notice you are performing some special handling in jobsdb, where you only include an error for the 1st job of the batch (not sure I like it). In that context, is this still necessary?

Reply:
>include an error for the 1st job of the batch

this felt simpler. I thought of introducing a jobsdb.JobBatch kind of struct, but it'd have been more work, and could be taken up at a later point considering I didn't any breaking issue going with the first job's UUID like I've done here.

Reply:
This detail definitely needs to be included in the josbdb method's doc, since one would naturally expect that all errors for all uuids would be returned, instead of the first one.

Reply:
yeah, realised that above. Adding that part in docs right now.